eam-light-backendweb/src/main/java/ch/cern/cmms/eamlightweb/equipment/EquipmentReplacementService.java

```diff
@@ -182,6 +182,35 @@ private String replaceEquipmentSwapping(InforContext inforContext, EquipmentRepl
         return replacement.getOldEquipment() + " was replaced by " + replacement.getNewEquipment();
     }
 
+    /**
+     * Recursively detaches dependent children from their positions if they are dependent on their parent asset
+     *
+     * @param parentEquipment
+     *            The uppermost parent asset of the hierarchy being replaced
+     */
+    private void detachDependentChildrenFromPosition(InforContext inforContext, Equipment parentEquipment) throws InforException {
+        List<EquipmentStructure> parentEquipmentChildren = getDirectChildren(parentEquipment);
+
+        for (EquipmentStructure child : parentEquipmentChildren) {
+            // Check for parent asset dependency
+            if (child.getDependent()) {
+                String childCode = child.getChildCode();
+                Equipment childEquipment = inforClient.getAssetService().readAsset(inforContext, childCode, null);
```

Comment:
We could, at the very least, batch the read asset op

Reply:
The `readAsset` call cannot benefit from `batchOperation` since it takes a parameter besides the `assetCode`, the `organization`. Unless you were thinking of something else I would say maybe we optimize later if needed?

Reply:
Except we are not even using the organization (`null` is literally being passed), which makes the organization come from the inforContext, which does not change. And even if we did somehow, the actual operation being performed would most often than not be for assets in the same organization (and in CERN's case, always); grouping the batch by organization would still be a performance improvement. So is it because we don't want to add the readAssetBatch  is not (yet) available or is there another reason I missed?

Reply:
Another thing I did not mention is that, at least according to the comments, this method `detaches dependent children from their positions if they are dependent on their parent asset`, are we ignoring the possibly dependent children positions on purpose?

Reply:
In the end there is a `readEquipmentBatch` that we can use. I've also added support for positions, but this issue still needs extra clarification so I am marking this as a WIP for now